src/parameters/mod.ts

```diff
@@ -69,34 +93,42 @@ export const ParameterVariable = <P extends ParameterDefinition>(
 ): ParameterVariableType<P> => {
   let param: ParameterVariableType<P> | null = null;
 
-  if (definition.type instanceof Object) {
-    param = ParameterVariable(
+  if (isCustomType(definition.type)) {
+    return ParameterVariable(
       namespace,
       paramName,
       definition.type.definition,
     );
-  } else if (definition.type === SchemaTypes.object) {
-    if ("properties" in definition) {
-      param = CreateTypedObjectParameterVariable(
+  }
+  switch (definition.type) {
```

Comment:
💭 Early on in our exploration we moved to this `switch` statement. Do we think that using a switch statement is still preferable considering we needed to add an if statement above?

Reply:
Probably doesn't make any sense anymore. Should I remove it and flip back to `if`s?

Reply:
Yeah that makes sense to me